### ISSUE TITLE
InjectRegister not working when disabling manifest

### ISSUE THREAD

**Issue by harshagarwal-mono** [vite-pwa/vite-plugin-pwa#660]
Even though i set injectRegister as null still it is auto Registering. How do i stop it ?

**Comment by userquin** [comment]
what do you mean? do you want to register the sw yourself? if so, set `injectRegister` to false  and do not use any virtual pwa module

**Comment by harshagarwal-mono** [comment]
Yes i want to register the sw myself. I have set the injectRegister: null in settings. And commented the code where virtual pwa module was used.

**Comment by userquin** [comment]
The `null` value is there, but it shouldn't, added changes to how we load options may have broken the `null` value. I'm going to change the jsdocs and deprecate the "null" value.

![imagen](https://github.com/vite-pwa/vite-plugin-pwa/assets/6311119/af595148-b6cc-458f-a2b0-e17e4f2a608a)

**Comment by harshagarwal-mono** [comment]
false option also not working. I have also checked with that. Keeping same in mind. I am using nuxt3. Can you suggest something to either debug or how do i get it fixed ?

**Comment by userquin** [comment]
Upps, if you using Nuxt you need to disable the plugin, it is registering the sw via `virtual:pwa-register/vue`: https://github.com/vite-pwa/nuxt/blob/main/src/runtime/plugins/pwa.client.ts#L55 setting `client.registerPlugin` to false.

I'm going to add an option to allow disable sw registration...

**Comment by harshagarwal-mono** [comment]
Thanks. This will help me to understand problem. Just one more thing. I didn;t get the documentation for periodicSyncForUpdate configuration. Is it possible if you could provude me that link. I am doing it manually right now.

**Comment by userquin** [comment]
Use `client.periodicSyncForUpdates` in seconds: https://github.com/vite-pwa/nuxt/blob/main/src/types.ts#L11C3-L11C25 in the docs it is in the Nuxt section: https://vite-pwa-org.netlify.app/frameworks/nuxt.html

**Comment by harshagarwal-mono** [comment]
Thanks for the help

**Comment by harshagarwal-mono** [comment]
Just one more thing. I have observed that if app is already offlineReady then it does not give me true value. Is it expected ?

**Comment by userquin** [comment]
> Just one more thing. I have observed that if app is already offlineReady then it does not give me true value. Is it expected ?.

It is only true when sw installed for first time.

**Comment by rschristian** [comment]
Just ran into this too. If you set `manifest: false`, this breaks `injectRegister` (in dev, anyhow). The docs state that `manifest: false` should only disable the automatic generation of a webmanifest, so I don't think the behavior here is correct.

```js
vitePWA({
    injectRegister: 'inline',
    manifest: false,  // comment me out to resume injection
    devOptions: {
        enabled: true,
    },
})
```

**Comment by userquin** [comment]
@rschristian can you file a new issue?

**Comment by rschristian** [comment]
Oh I can do, sorry. Must've misread this issue when skimming.

**Comment by userquin** [comment]
The problem seems to be only in dev server, when building seems to generated. @rschristian can you confirm this?

**Comment by rschristian** [comment]
Yes, I can confirm that. Works exactly as expected using the production build.

**Comment by userquin** [comment]
> Yes, I can confirm that. Works exactly as expected using the production build.

Thx.

The fix is quite easy, I'm going to release a new patch version `0.19.4` in a few minutes, I'll ping you here when released.

**Comment by userquin** [comment]
@rschristian released 0.19.4
